include/aspect/utilities.h

```diff
@@ -110,6 +114,14 @@ namespace aspect
      * @param[in] property_name A name that identifies the type of property
      *   that is being parsed by this function and that is used in generating
      *   error messages if the map does not conform to the expected format.
+     * @param [in] allow_subentries If true allow having multiple subentries
+     *   for each key. If false only allow a single entry per key. In either
+     *   case each key is only allowed to appear once.
+     * @param [out] subentry_structure If handed over this
```

Comment:
To me, this name does not seem as descriptive as it could be. What about something like `values_per_key` or `n_values_per_key`? That would also be consistent with the naming of the other input parameters.